src/main/java/org/sonarsource/sonarlint/ls/clientapi/SonarLintVSCodeClient.java

```diff
@@ -254,9 +260,10 @@ private HashMap<String, ServerConnectionSettings> getCurrentConnections(AssistCr
 
   @Override
   public AssistBindingResponse assistBinding(AssistBindingParams params, CancelChecker cancelChecker) {
+    workspaceFoldersManager.updateAnalysisReadiness(Set.of(params.getConfigScopeId()), false);
     return client.assistBinding(params)
-      .thenCompose(response -> bindingManager.getUpdatedBindingForWorkspaceFolder(URI.create(response.getConfigurationScopeId())))
```

Comment:
the whole method ` bindingManager.getUpdatedBindingForWorkspaceFolder ` can be removed, it's only referenced from tests now.